api/v3/PaymentProcessor.php

```diff
@@ -158,6 +159,11 @@ function _civicrm_api3_payment_processor_pay_spec(&$params) {
     'title' => ts('Amount to pay'),
     'type' => CRM_Utils_Type::T_MONEY,
   ];
+  $params['contribution_id'] = [
```

Comment:
Why is this required? The goal is to action a payment. Is it to ensure the info is always available to the processor & force people into the create-pending-contribution first flow?

Reply:
See #15477